packages/models-library/src/models_library/projects.py

```diff
@@ -1,153 +1,107 @@
-    delete: bool = Field(..., description="gives deletion rights")
-
-    class Config:
-        extra = Extra.forbid
-
-
-class Owner(BaseModel):
-    user_id: PositiveInt = Field(
-        ...,
-        description="Owner's identifier when registered in the user's database table",
-        example=[2],
-    )
-    first_name: str = Field(..., description="Owner first name", example=["John"])
-    last_name: str = Field(..., description="Owner last name", example=["Smith"])
-
-    class Config:
-        extra = Extra.forbid
-
-
-class ProjectLocked(BaseModel):
-    value: bool = Field(
-        ..., description="True if the project is locked by another user"
-    )
-    owner: Optional[Owner] = Field(None, description="The user that owns the lock")
-
-    class Config:
-        extra = Extra.forbid
-
-
-class ProjectRunningState(BaseModel):
-    value: RunningState = Field(
-        ..., description="The running state of the project", example=["STARTED"]
-    )
-
-    class Config:
-        extra = Extra.forbid
 
 
-class ProjectState(BaseModel):
-    locked: ProjectLocked = Field(..., description="The project lock state")
-    state: ProjectRunningState = Field(..., description="The project running state")
-
-    class Config:
-        extra = Extra.forbid
+class Workbench(BaseModel):
+    __root__ = Dict[NodeID_AsDictKey, Node]
 
 
 class Project(BaseModel):
     uuid: ProjectID = Field(
         ...,
         description="project unique identifier",
-        example=[
+        examples=[
             "07640335-a91f-468c-ab69-a374fa82078d",
             "9bcf8feb-c1b1-41b6-b201-639cd6ccdba8",
         ],
     )
+
+    # Description of the project
     name: str = Field(
-        ..., description="project name", example=["Temporal Distortion Simulator"]
+        ..., description="project name", examples=["Temporal Distortion Simulator"]
     )
     description: str = Field(
         ...,
         description="longer one-line description about the project",
-        example=["Dabbling in temporal transitions ..."],
+        examples=["Dabbling in temporal transitions ..."],
     )
+    thumbnail: HttpUrl = Field(
+        ...,
+        description="url of the project thumbnail",
+        examples=["https://placeimg.com/171/96/tech/grayscale/?0.jpg"],
+    )
+
+    # Ownership and Access  (SEE projects_access.py)
     prjOwner: EmailStr = Field(..., description="user email")
     accessRights: Dict[GroupID, AccessRights] = Field(
         ...,
         description="object containing the GroupID as key and read/write/execution permissions as value",
     )
+
+    # Timestamps   TODO: should we use datetime??
     creationDate: str = Field(
         ...,
         regex=DATE_RE,
         description="project creation date",
-        example=["2018-07-01T11:13:43Z"],
+        examples=["2018-07-01T11:13:43Z"],
     )
     lastChangeDate: str = Field(
         ...,
         regex=DATE_RE,
```

Comment:
I find a bit weird that you need to define a DATE_RE and that nothing is already thought out in the native python... but maybe they are as bad as C++ in that aspect?

Reply:
pydantic supports datetime. In #1976 we could review these types